src/components/Card/Card.tsx

```diff
@@ -11,18 +12,52 @@ function Card({
   flip: boolean;
   onClick: () => void;
 }) {
+  const cardholderRef = React.useRef<HTMLDivElement>(null);
+
+  React.useEffect(() => {
+    if (!cardholderRef.current) {
+      return;
+    }
+    const element = cardholderRef.current;
+    let entryPoint = { offsetX: 0, offsetY: 0 };
+    function track(event: MouseEvent) {
+      element.style.transform = `translate(${
+        (event.offsetX - entryPoint.offsetX) * 0.02
+      }px, ${(event.offsetY - entryPoint.offsetY) * 0.02}px)`;
+    }
+
+    function startTracking(event: MouseEvent) {
+      entryPoint = { offsetX: event.offsetX, offsetY: event.offsetY };
+      element.addEventListener("mousemove", track);
+    }
+
+    function stopTracking() {
+      element.removeEventListener("mousemove", track);
+      element.removeAttribute("style");
+    }
+
+    element.addEventListener("mouseenter", startTracking);
+    element.addEventListener("mouseleave", stopTracking);
```

Comment:
Can the same be done this way? Will it still work the same way?

```tsx
...
const handleMouseMove: React.MouseEventHandler<HTMLDivElement> = (event) => {
    const el = event.target as HTMLDivElement;
    const offsetX = event.nativeEvent.offsetX * 0.02;
    const offsetY = event.nativeEvent.offsetY * 0.02;

    el.style.transform = `translate(${offsetX}px, ${offsetY}px)`;
  };

  const handleMouseLeave: React.MouseEventHandler<HTMLDivElement> = (event) => {
    const el = event.target as HTMLDivElement;

    el.removeAttribute("style");
  };

return (
    <div
      className={styles.cardholder}
      onClick={onClick}
      onMouseMove={handleMouseMove}
      onMouseLeave={handleMouseLeave}
    >
...
```

if yes, it looks more concise and simpler than useEffect

Reply:
This does work, but not exactly right. It only works as intended if your cursor enters the div at [0, 0] coordinates. Otherwise, it jerks the whole div in the direction of your cursor. My guess is that it is happening because you don't account for the entryPoint in this implementation.

Reply:
But you can make it work by adding entry point?

Reply:
I mean, entry point is an important detail that I missed, but the point was to show another possible solution. The logic stays the same, but implementation looks more straightforward to me. You may see it differently, ofc

Reply:
I tried introducing entryPoint into this approach, but it still works weird. My guess: it has something to do with how I store the entryPoint (in state). There's obviously something wrong with how I do it, but I can't put my finger on what it is exactly. That's basically why I went down the useEffect road before, as it allowed me to have a simple variable scoped to the same callback.

Reply:
I don't fully understand what's going on inside this useEffect, which is why it's not easy to implement it differently. Do you? I find it hard to get through all of these add/remove event listeners wrapped inside each other. 
I think it's important to first fully understand what it does, then it will be clear what you need to do

Reply:
It's pretty straightforward, really.
1. MouseEnter listener. As cursor enters the div, you record the entry point and enable tracking by adding the MouseMove listener.
2. MouseMove listener. With each mouse movement you recalculate the offset and apply the style to the div.
3. MouseLeave listener. As cursor leaves the div, you delete the MouseMove listener and reset the div's style.

Reply:
It's not so straightforward when you read the code, that's why I didn't like the initial solution. But I see that capturing starting point in the alternative solution is problematic - onMouseMove shoots with the old state at first as it doesn't wait for onMouseEnter to set new state. 
Let's leave it as is for now.